import express from 'express';
import cors from 'cors';
import dotenv from 'dotenv';
import { authRouter } from './routes/authRoutes.js';
import { dashboardRouter } from './routes/dashboardRoutes.js';
import { indicadoresRouter } from './routes/indicadoresRoutes.js';
import { protect } from './middleware/authMiddleware.js';

dotenv.config();
const app = express();

app.use(cors());
app.use(express.json());

app.use('/api/auth', authRouter);
app.use('/api/dashboard', protect, dashboardRouter);
app.use('/api/indicadores', protect, indicadoresRouter);

app.use((err, req, res, next) => {
  console.error("ERROR FINAL DETECTADO:", err);
  res.status(500).json({ message: 'Error interno del servidor', error: err.message });
});

const PORT = process.env.PORT || 3001;
app.listen(PORT, () => {
  console.log(`ðŸš€ Servidor API corriendo en http://localhost:${PORT}`);
});